tools/createSovereignGenesis/create-sovereign-genesis.ts

```diff
@@ -145,6 +188,56 @@ async function main() {
         outWETHAddress = wethObject.address;
     }
 
+    // check preMintAddress is an address
+    if (createGenesisSovereignParams.setPreMintAccount === true) {
+        logger.info('Add preMintAccount');
+        if (ethers.isAddress(createGenesisSovereignParams.preMintAccount.address) == false) {
+            logger.error('preMintAccount.address: not a valid address');
+            throw new Error('preMintAccount.address: not a valid address');
+        }
+
+        // add preMintAccount.address & preMintAccount.balance
+        finalGenesis.genesis.push({
+            accountName: 'preMintAccount',
+            balance: BigInt(createGenesisSovereignParams.preMintAccount.balance).toString(),
+            address: createGenesisSovereignParams.preMintAccount.address,
+        });
+    }
+
+    // set timelock storage
+    if (createGenesisSovereignParams.setTimelockParameters === true) {
+        logger.info('Add timelockParameters');
+        const timelockContractInfo = finalGenesis.genesis.find(function (obj) {
+            return obj.contractName === 'PolygonZkEVMTimelock';
+        });
+
+        const storageTimelock = initializeTimelockStorage(
+            createGenesisSovereignParams.timelockParameters.minDelay,
+            createGenesisSovereignParams.timelockParameters.adminAddress,
+            timelockContractInfo.address,
+        );
+
+        timelockContractInfo.storage = storageTimelock;
+    }
+
+    // regenerate root with the zkEVM root
+    const poseidon = await getPoseidon();
+    const {F} = poseidon;
+
+    const zkEVMDB = await ZkEVMDB.newZkEVM(
+        new MemDB(F),
+        poseidon,
+        [F.zero, F.zero, F.zero, F.zero],
+        [F.zero, F.zero, F.zero, F.zero],
+        finalGenesis.genesis,
+        null,
+        null,
+        createGenesisSovereignParams.chainID
+    );
+
+    // update genesis root
+    finalGenesis.root = smtUtils.h4toString(zkEVMDB.getCurrentStateRoot());
+
     // Populate final output
     outputJson.network = hardhatArguments.network;
```

Comment:
Maybe add also info about the timelock admin/delay to the output (maybe no need about preminted account because is explicit in the genesis)

Reply:
yep!